sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -497,7 +497,10 @@ object DataSourceStrategy {
         Some(sources.IsNotNull(a.name))
 
       case expressions.And(left, right) =>
-        (translateFilter(left) ++ translateFilter(right)).reduceOption(sources.And)
+        for {
```

Comment:
Let's add a small comment like the PR you pointed out.

Reply:
Sure. Will do. Thanks.

Reply:
Yeah. Follow what @yhuai  wrote in the PR https://github.com/apache/spark/pull/10362/files

Reply:
Thanks. Just did that as you suggested.